core/job/service/job_service_test.go

```diff
@@ -1874,21 +1923,57 @@ func TestJobService(t *testing.T) {
 			specC, _ := job.NewSpecBuilder(jobVersion, "job-C", "sample-owner", jobSchedule, jobWindow, jobTask).Build()
 			specs := []*job.Spec{specA, specB, specC}
 
-			jobA := job.NewJob(sampleTenant, specA, "table-A", []job.ResourceURN{"table-C"})
+			jobA := job.NewJob(sampleTenant, specA, "table-A", []job.ResourceURN{"table-Z"})
 			jobB := job.NewJob(sampleTenant, specB, "table-B", []job.ResourceURN{"table-A"})
 			jobC := job.NewJob(sampleTenant, specC, "table-C", []job.ResourceURN{"table-B"})
+
+			jobsWithUpstreams := []*job.WithUpstream{
+				job.NewWithUpstream(
+					jobB,
+					[]*job.Upstream{
+						job.NewUpstreamUnresolvedInferred(job.ResourceURN("table-A")),
```

Comment:
let's add or combine with static upstream to also test the cyclic due static upstream